src/core/document.js

```diff
@@ -1020,6 +1021,26 @@ class PDFDocument {
     );
   }
 
+  get hasJSActions() {
+    return shadow(
+      this,
+      "hasJSActions",
+      this.fieldObjects.then(objects => {
+        if (objects === null) {
+          return false;
+        }
+        for (const object of Object.values(objects)) {
+          for (const obj of object) {
```

Comment:
It might be more clear to rename `objects` -> `fieldObjects`, `object` -> `fieldObject` and `obj` -> `object`.